src/common/locallaplaciancl.c

```diff
@@ -88,11 +88,11 @@ dt_local_laplacian_cl_t *dt_local_laplacian_init_cl(
   g->shadows = shadows;
   g->highlights = highlights;
   g->clarity = clarity;
-  g->dev_padded = (cl_mem *)calloc(max_levels, sizeof(cl_mem *));
-  g->dev_output = (cl_mem *)calloc(max_levels, sizeof(cl_mem *));
-  g->dev_processed = (cl_mem **)calloc(num_gamma, sizeof(cl_mem **));
+  g->dev_padded = calloc(max_levels, sizeof(cl_mem));
+  g->dev_output = calloc(max_levels, sizeof(cl_mem));
+  g->dev_processed = calloc(num_gamma, sizeof(cl_mem *));
   for(int k=0;k<num_gamma;k++)
-    g->dev_processed[k] = (cl_mem *)calloc(max_levels, sizeof(cl_mem *));
```

Comment:
These erroneous memory allocations above work because of the accidental relation `sizeof(cl_mem *) == sizeof(cl_mem)`.